src/Microsoft.ML.AutoML/API/RecommendationExperiment.cs

```diff
@@ -0,0 +1,74 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using Microsoft.ML.Data;
+using Microsoft.ML.Trainers;
+
+namespace Microsoft.ML.AutoML
+{
+    public enum RecommendationExperimentScenario
+    {
+        MF,
+    }
+
+    public sealed class RecommendationExperimentSettings : ExperimentSettings
+    {
+        public RecommendationExperimentScenario Scenerio { get; set; }
+
+        public string MatrixColumnIndexColumnName { get; set; }
+
+        public string MatrixRowIndexColumnName { get; set; }
+
+        // We can use RegressionMetric as evaluation Metric
+        public RegressionMetric OptimizingMetric { get; set; }
+
+        public ICollection<RecommendationTrainer> Trainers { get; }
+
+        public RecommendationExperimentSettings(RecommendationExperimentScenario scenario, string columnIndexName, string rowIndexName)
+            : this()
+        {
+            if(scenario == RecommendationExperimentScenario.MF)
+            {
+                AutoCatalog.ValuePairs[nameof(MatrixFactorizationTrainer.Options.MatrixColumnIndexColumnName)] = columnIndexName;
+                AutoCatalog.ValuePairs[nameof(MatrixFactorizationTrainer.Options.MatrixRowIndexColumnName)] = rowIndexName;
+                return;
+            }
+            throw new NotImplementedException();
+        }
+
+        private RecommendationExperimentSettings()
+        {
+            OptimizingMetric = RegressionMetric.RSquared;
```

Comment:
I think this depends on which style of recommendation we are using. If I recall, we had a method which produces binary classification metrics.